utils/logging/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+//! The second one, client-side by implementing check against Record Level field.
+//! Each module may have its own Logger or share one.
+
+use core::{cmp, fmt};
+
+static LOG_LEVEL_NAMES: [&str; 6] = ["GOSSIP", "TRACE", "DEBUG", "INFO", "WARN", "ERROR"];
+
+/// An enum representing the available verbosity levels of the logger.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub enum Level {
+    /// Designates extremely verbose information, including gossip-induced messages
+    Gossip,
+    /// Designates very low priority, often extremely verbose, information
+    Trace,
+    /// Designates lower priority information
+    Debug,
+    /// Designates useful information
+    Info,
+    /// Designates hazardous situations
+    Warn,
+    /// Designates very serious errors
+    Error,
+}
+
+impl PartialOrd for Level {
+    #[inline]
+    fn partial_cmp(&self, other: &Level) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+
+    #[inline]
+    fn lt(&self, other: &Level) -> bool {
+        (*self as usize) < *other as usize
+    }
+
+    #[inline]
+    fn le(&self, other: &Level) -> bool {
+        *self as usize <= *other as usize
+    }
+
+    #[inline]
+    fn gt(&self, other: &Level) -> bool {
+        *self as usize > *other as usize
+    }
+
+    #[inline]
+    fn ge(&self, other: &Level) -> bool {
+        *self as usize >= *other as usize
+    }
+}
+
+impl Ord for Level {
+    #[inline]
+    fn cmp(&self, other: &Level) -> cmp::Ordering {
+        (*self as usize).cmp(&(*other as usize))
+    }
+}
+
+impl fmt::Display for Level {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.pad(LOG_LEVEL_NAMES[*self as usize])
+    }
+}
+
+impl Level {
+    /// Returns the most verbose logging level.
+    #[inline]
+    pub fn max() -> Level {
+        Level::Gossip
+    }
+}
+
+/// A Record, unit of logging output with Metadata to enable filtering
+/// Module_path, file, line to inform on log's source
+#[derive(Clone, Debug)]
+pub struct Record<'a> {
+    /// The verbosity level of the message.
+    pub level: Level,
+    /// The message body.
+    pub args: fmt::Arguments<'a>,
+    /// The module path of the message.
+    pub module_path: &'static str,
+    /// The source file containing the message.
+    pub file: &'static str,
+    /// The line containing the message.
+    pub line: u32,
+}
+
+impl<'a> Record<'a> {
+    /// Returns a new Record.
+    #[inline]
+    pub fn new(
+        level: Level,
+        args: fmt::Arguments<'a>,
+        module_path: &'static str,
+        file: &'static str,
+        line: u32,
+    ) -> Record<'a> {
+        Record {
+            level,
+            args,
+            module_path,
+            file,
+            line,
+        }
+    }
+}
+
+/// A trait encapsulating the operations required of a logger
+pub trait Logger {
+    /// Logs the `Record`
+    fn log(&self, record: &Record);
+}
```

Comment:
I would also implementd a logger so that we can use it in the roles without reimplemting it everythime something like:
```rust
enum Destination {
    Std,
    Costum {path_info: String, path_err: String },
}

pub struct MyLogger {
    destination: Destination
}

impl Logger for MyLogger {
    fn log(&self, record: &Record) {
        match self.destination {
            Destination::Std => {
                match record.level {
                    Level::Info => println!("{}",record.args),
                    //Level::Error => printerr!("{}",record.args),
                    _ => (),
                }
            }
            Destination::Costum { path_info, path_err } => todo!(),
            }
        }

    }
}

```